grpc-fixture/fixture/fixture_interceptor.go

```diff
@@ -10,115 +11,141 @@ import (
-						level1Nesting, ok := message.message.Message.(map[string]interface{})
-						if !ok {
-							return nil
+						if info.FullMethod == "/s12.tasks.v1.ActionsService/GetAction" {
+							//Override the action ID from dump on ID that's coming from client
+							level1Nesting, ok := message.message.Message.(map[string]interface{})
+							if !ok {
+								return nil
+							}
+							actionMap, ok := level1Nesting["action"].(map[string]interface{})
+							if !ok {
+								return nil
+							}
+							taskMap, ok := actionMap["task"].(map[string]interface{})
+							if !ok {
+								return nil
+							}
+							taskMap["taskId"] = taskId
+
+							msgBytes, err = f.encoder.Encode(info.FullMethod, message.message)
+							if err != nil {
+								return err
+							}
 						}
-						incidentMap, ok := level1Nesting["incident"].(map[string]interface{})
-						if !ok {
-							return nil
+						if info.FullMethod == "/s12.tasks.v1.IncidentsService/GetIncident" {
+							level1Nesting, ok := message.message.Message.(map[string]interface{})
+							if !ok {
+								return nil
+							}
+							incidentMap, ok := level1Nesting["incident"].(map[string]interface{})
+							if !ok {
+								return nil
+							}
+							taskMap, ok := incidentMap["task"].(map[string]interface{})
+							if !ok {
+								return nil
+							}
+							taskMap["taskId"] = taskId
+
+							msgBytes, err = f.encoder.Encode(info.FullMethod, message.message)
+							if err != nil {
+								return err
+							}
 						}
-						taskMap, ok := incidentMap["task"].(map[string]interface{})
-						if !ok {
-							return nil
+						sendMsgErr := ss.SendMsg(msgBytes)
+						if sendMsgErr != nil {
+							return sendMsgErr
 						}
-						taskMap["taskId"] = taskId
 
-						encodeErr := error(nil)
-						msgBytes, encodeErr = f.encoder.Encode(info.FullMethod, message.message)
-						if encodeErr != nil {
-							return encodeErr
-						}
-					}
-					sendMsgErr := ss.SendMsg(msgBytes)
-					if sendMsgErr != nil {
-						return sendMsgErr
-					}
-					// recurse deeper into the tree
-					message.called = true
-					messageTreeNode = message
-					if len(messageTreeNode.nextMessages) == 0 {
-						messageTreeNode.called = true
-						// end of the exchange
-						return nil
-					}
-					// found the first server message so break
-					break
-				} else {
-					// wait for a client message and then proceed based on its contents
-					var receivedMessage []byte
-					err := ss.RecvMsg(&receivedMessage)
-					if err != nil {
-						return err
-					}
-					if info.FullMethod == "/s12.tasks.v1.ActionsService/GetAction" {
-						receivedMessageStructure := internal.Message{
-							MessageOrigin: internal.ClientMessage,
-							RawMessage:    receivedMessage,
-							Message:       nil,
-							Timestamp:     time.Time{},
+						// recurse deeper into the tree
+						message.called = true
+						gc.ChangeState(true)
```

Comment:
not required here

Reply:
do we still require message.called be set to true or nah?